src/components/select/select.component.ts

```diff
@@ -377,6 +376,15 @@ export class SelectComponent<T> implements OnInit, OnChanges, OnDestroy, Control
             this._userInput = true;
             this._dropdownOpen = true;
         }
+
+        setTimeout(() => {
+            let target = event.target as HTMLInputElement;
+            if (event.keyCode === DELETE && target.value === '' && this.allowNull) {
+                this.valueChange.next(null);
+            }
+        }, 10);

```

Comment:
I know this probably isn't an ideal solution for this (it will fire if you keep pressing DELETE for example), the problem I am having is getting the current value of the text in the input as the _input$ observables fire after this is called. Like you mentioned, perhaps the way this component handles this input would need changed?

Reply:
According to the provided plunker, the issue is that null is emitted when the `value` input is set programmatically, so it's not really anything to do with key handling. If you trace what happens when the setter for `value` is called you will find that it ultimately calls `valueChange`. What you need to do is eliminate any calls of valueChange which come from changes to the component's inputs. It should only be called when the user does something (keydown, click, blur, etc).

I think this will probably be a complicated task. Since the component relies heavily on observables, you may need to change them to carry a structured object which includes a flag indicating whether the change from from a user action or not.

Reply:
Correction to the above comment, I do see what you are getting at with this fix. I'm not sure if it can be made to work without the kind of systematic change I mentioned above however. If you carry a flag along with the string value of input you can decide whether to emit the event. Hopefully that makes sense.